### ISSUE TITLE
Speedup and reduced memory usage for ashikhmin

### ISSUE THREAD

**Issue by heckflosse** [LuminanceHDR/LuminanceHDR#87]
I created a new [branch](https://github.com/heckflosse/LuminanceHDR/tree/ashikhmin_speedup) in my fork with a speedup and reduced memory usage for ashikhmin.

I used a 24 Mpixel file for my tests. Here are the timings:
```
                 before                    after
Eqn 2           16100 ms                  1975 ms
Eqn 4           20460 ms                  2015 ms
```

Memory usage is reduced by `width * height * 4 byte`

**Comment by Benitoite** [comment]
That's six speedups merged on Test1 of my fork and built for mac, all working great!
https://filebin.net/sk8qnnj53i8hgobb/Luminance_HDR_2.5.2-SNAPSHOT-Test1f.dmg

**Comment by heckflosse** [comment]
Fixed with 0b394f7
